network/discovery.go

```diff
@@ -24,46 +23,66 @@ var discProto = "/disc/0.1"
 
 const defaultBucketSize = 20
 
-// referencePeer is a representation of the peer the node is requesting for new peers
 type referencePeer struct {
 	id     peer.ID
 	stream interface{}
 }
 
-type referencePeers []*referencePeer
+type referencePeers struct {
+	mux   sync.RWMutex
+	peers []*referencePeer
+}
 
 func (ps *referencePeers) find(id peer.ID) *referencePeer {
-	for _, p := range *ps {
+	ps.mux.RLock()
+	defer ps.mux.RUnlock()
+
+	for _, p := range ps.peers {
 		if p.id == id {
 			return p
 		}
 	}
 	return nil
 }
 
+func (ps *referencePeers) getRandomPeer() *referencePeer {
+	ps.mux.RLock()
+	defer ps.mux.RUnlock()
+
+	l := len(ps.peers)
+	if l == 0 {
+		return nil
+	}
+	return ps.peers[rand.Intn(l)]
```

Comment:
Where do we seed this random number generator?

Reply:
@zivkovicmilos https://github.com/0xPolygon/polygon-sdk/blob/82cf7177cb9f48c9d06fcf3ab921e28375437e58/network/discovery.go#L18-L20

Reply:
This method is never called

Reply:
@zivkovicmilos https://golangbyexample.com/init-function-golang/